### PR TITLE
Pyvisapy

### PR DESCRIPTION
**Pull request by Schneider1**
Modifikations to make qkit work with pyvisapy (and newer versions of pyvisa)
It works with pyvisapy and should not harm for NI-Visa installations, but the latter is not tested.

### CODE REVIEW

File: qkit/drivers/Keysight_VNA_E5080B.py [hunk 1]
```diff
@@ -870,7 +871,7 @@ def post_measurement(self):
         """
         Bring the VNA back to a mode where it can be easily used by the operator.
         """
-        self.set_sweep_mode("hold")
```
Comment (TimWolz) [inline qkit/drivers/Keysight_VNA_E5080B.py:877]: Isn't hold better to not accidently excite/heat your sample? Also, one can directly see in the lab if a measurement is running or not

File: qkit/drivers/Agilent_VNA_E5071C.py [hunk 2]
```diff
@@ -724,15 +741,15 @@ def start_measurement(self):
         This function is called at the beginning of each single measurement in the spectroscopy script.
         Here, it resets the averaging.  
         '''
-        self._visainstrument.write(':TRIG:AVER ON')
+        self.write(':TRIG:AVER ON')
```
Comment (TimWolz) [inline qkit/drivers/Agilent_VNA_E5071C.py:744]: If I remember correctly, @MartinSpiecker had some problems with that command. Their (new) VNAs were ignoring the trigger afterwards.

File: qkit/drivers/Keysight_VNA_E5071C.py [hunk 3]
```diff
@@ -916,31 +839,44 @@ def do_set_sweep_type(self,swtype):
...
+            return self._visainstrument.query_binary_values(msg,datatype=dtype,container=numpy.array)
      
     def pre_measurement(self):
         '''
         Set everything needed for the measurement
         '''
-        self._visainstrument.write(":TRIG:SOUR BUS")#Only wait for software triggers
-        self._visainstrument.write(":TRIG:AVER ON")# Tell the instrument to do the specific number of averages on every trigger.
+        self.write(":TRIG:SOUR BUS")#Only wait for software triggers
+        self.write(":TRIG:AVER ON")# Tell the instrument to do the specific number of averages on every trigger.
```
Comment (TimWolz) [inline qkit/drivers/Keysight_VNA_E5071C.py:870]: If I remember correctly, @MartinSpiecker had some problems with that command. Their (new) VNAs were ignoring the trigger afterwards.

**Review by TimWolz — COMMENTED** [review]
import visa is apparently deprecated and should be used as import pyvisa as visa.

Also, we have to have an option to specify if pyvisa-py or NI-VISA should be used. Sth like 
if qkit.cfg['visa_backend'] = 'NI-Visa: rm = visa.ResourceManager()
elif qkit.cfg['visa_backend']='py': rm = visa.ResourceManager('@py') ?

File: qkit/drivers/Keysight_VNA_E5080B.py (hunk 1 above)
Comment (Schneider1) [reply qkit/drivers/Keysight_VNA_E5080B.py:877]: I guess that's a matter of personal taste. But we should have it equal on all devices/drivers. 

The primary intention of this command was that you don't have to do anything to be able to measure at the device directly after a measurement command. So it has to be running. But I see your point.

File: qkit/drivers/Keysight_VNA_E5071C.py (hunk 3 above)
Comment (Schneider1) [reply qkit/drivers/Keysight_VNA_E5071C.py:870]: If you have issues, please make a new issue or create a fix.

**Comment by TimWolz** [comment]
Oh I'm sorry. I totally forgot to mention I tested your code with the new Keysight VNA and mw source and all worked well.

**Review by TimWolz — APPROVED** [review]
Cool! Thanks a lot for your work, here

**Pull request merged**